code/modules/client/client_procs.dm

```diff
@@ -336,6 +336,36 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 				qdel(src)
 				return
 
+	if(CONFIG_GET(flag/panic_bunker) && CONFIG_GET(flag/panic_bunker_discord_require))
+		if(!SSdbcore.Connect())
+			var/msg = "Database connection failure. Key [key] not checked for Discord account requirement."
+
+			if(!CONFIG_GET(flag/sql_enabled))
+				msg += "\nDB IS NOT ENABLED - THIS IS NOT A BUG\nDiscord account links cannot be checked without a database!"
+
+			log_world(msg)
+			message_admins(msg)
+		else
+			if(!discord_is_link_valid(ckey))
+				var/discord_otp = discord_get_or_generate_one_time_token_for_ckey(ckey)
+
+				//These need to be immediate because we're disposing of the client the second we're done with this.
+				to_chat_immediate(src, span_danger(CONFIG_GET(string/panic_bunker_discord_register_message)))
+				to_chat_immediate(src, span_boldnotice("Your One-Time-Password is: [discord_otp]"))
+				to_chat_immediate(src, span_userdanger("DO NOT SHARE THIS OTP WITH ANYONE"))
+				var/discord_prefix = CONFIG_GET(string/discordbotcommandprefix)
+				to_chat_immediate(src, span_notice("To link your Discord account, head to the Discord Guild and paste the following message:<hr/><code>[discord_prefix]verify [discord_otp]</code><hr/>\n"))
```

Comment:
I know that 'guild' is the actual name for discord servers, but most people probably won't recognise the term.
"head to the Discord Server" would probably be better to prevent any confusion: (Same with the other instances of "discord guild" as well)
```suggestion
				to_chat_immediate(src, span_notice("To link your Discord account, head to the Discord Server and paste the following message:<hr/><code>[discord_prefix]verify [discord_otp]</code><hr/>\n"))
```